### ISSUE TITLE
In Java, specify empty string results in undefined makes Cloudfront unusable

### ISSUE THREAD

**Issue by leepa** [awslabs/aws-cdk#815]
I if do:

```
        new CloudFrontWebDistribution(this, "CloudFront", CloudFrontWebDistributionProps.builder()
                .withDefaultRootObject("")
```

This will result in:

```
DefaultRootObject: index.html
```

This is obviously undesired behaviour and there doesn't appear to be a way to work around it.

**Comment by leepa** [comment]
I have resorted to using the L1 `DistributionResource` directly for now to work around the issue.

**Comment by rix0rrr** [comment]
From the sound of it, that'll be an

```ts
if (property)
```

Which should have been an

```ts
if (property !== undefined)
```

Somewhere in that construct.

**Comment by leepa** [comment]
It's not - in the code it uses === undefined correctly.

**Comment by rix0rrr** [comment]
Well huh. Wonder if it's a jsii bug somewhere.

**Comment by rix0rrr** [comment]
Proof: https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-cloudfront/lib/web_distribution.ts#L509

@RomainMuller @eladb ?

**Comment by RomainMuller** [comment]
I'll try to have a look at this one soon-ish. May be `jsii` turning a blank string into `undefined` before shipping it to the `node` process, but won't know for sure before I dive deeper.

**Comment by RomainMuller** [comment]
Okay - this is fixed by #801, which is not yet released (aka 0.10.0 doesn't have the fix). The next release will have this problem solved.

**Comment by RomainMuller** [comment]
To elaborate a bit more - this particular fix slipped right *after* we published the `npm`, `maven` and `NuGet` package, but we had issues publishing the release & this prompted us to re-tag on a later commit. This ONE particular fix is under the `v0.10.0` tag, but is only included in the GitHub release, and not anywhere else.
